### PR TITLE
cherry-pick changes from feature/ase_plugin to feature/ase_plugin2

### PR DESCRIPTION
**Pull request by lzhan55**
* feature/ase_plugin2 was just branched off from master today
* cmake .. -DCPACK_GENERATOR=DEB && make package_deb passed

### CODE REVIEW

File: libopae/plugins/ase/api/src/init.c [hunk 1]
```diff
@@ -0,0 +1,45 @@
+// Copyright(c) 2017-2018, Intel Corporation
```
Comment (tswhison) [inline libopae/plugins/ase/api/src/init.c]: 2017-2019

File: libopae/plugins/ase/api/src/plugin.c [hunk 2]
```diff
@@ -0,0 +1,188 @@
+// Copyright(c) 2018, Intel Corporation
```
Comment (tswhison) [inline libopae/plugins/ase/api/src/plugin.c]: 2018-2019 or just 2019, if it was created this year.

File: libopae/plugins/ase/scripts/generate_ase_environment.py [hunk 3]
```diff
@@ -60,8 +60,8 @@
 from sets import Set
 import shutil
 
-reload(sys)
-sys.setdefaultencoding('utf8')
+# reload(sys)
+# sys.setdefaultencoding('utf8')
```
Comment (tswhison) [inline libopae/plugins/ase/scripts/generate_ase_environment.py]: Need this enabled?

File: libopae/plugins/ase/api/src/plugin.c (hunk 2 above)
Comment (lzhan55) [reply libopae/plugins/ase/api/src/plugin.c]: will fix

File: libopae/plugins/ase/api/src/init.c (hunk 1 above)
Comment (lzhan55) [reply libopae/plugins/ase/api/src/init.c]: will fix

File: libopae/plugins/ase/scripts/generate_ase_environment.py (hunk 3 above)
Comment (lzhan55) [reply libopae/plugins/ase/scripts/generate_ase_environment.py]: it might be needed in some environment. For now I will remove it.

**Pull request merged**